### DISCUSSION TITLE
Is there a way to get an email's subject/body directly from an IDLE response?

### DISCUSSION THREAD

**Discussion by rynlu** [mjs/imapclient#436] [Q&A]
I was having trouble finding a way to get an email's associated data from the IDLE response and was wondering if this wasn't supported at all and I had to do a fetch command or if I was just missing something. Any help is appreciated, thanks!

**Comment by mjs** [comment]
I'm afraid not. IDLE responses are mainly useful for telling you that something has happened but they you need to run a search or fetch to find out what changed.

**Reply by ghost** [reply]
Say, IDLE is sending something like (131231, b'EXISTS'). What is the correct way to fetch this message 132231?
response = server.fetch( [131231], ['RFC822'] ) does not seem to be the correct way.

**Reply by octo-weilian** [reply]
An IDLE response with an EXISTS tag returns with a message sequence number. That is the number of messages in your folder. To directly fetch a message using a sequence number you can disable the `use_uid `argument (see example below). However, note that sequence numbers are not persistent. 

```
#connect to client  
IMAPClient('imap.mailserver.com', use_uid=False)
server.login('someuser', 'somepassword')
inbox= server.select_folder('INBOX')
```

```
#fetch a single message
seq, data = next(iter(server.fetch(131231, ['RFC822']).items()))
```

**Comment by mjs** [comment]
As per https://github.com/mjs/imapclient/discussions/471, the EXISTS response isn't giving you a message sequence number, just a count of messages. You can't use the IDLE EXISTS response like this.
